### ISSUE TITLE
Replace explicit slf4j dependency by JDK 9 System.Logger usage

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14509]
Starting with JDK 9, the JDK finally has its own facade to redirect logging calls to custom logger frameworks:
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/System.Logger.html

If we start using that in `JooqLogger`, then we can remove the slf4j dependency, allowing users to configure their own `System.Logger` implementation just like before.

The Java 8 distribution would continue depending on slf4j for that purpose.

**Comment by lukaseder** [comment]
Duplicate of https://github.com/jOOQ/jOOQ/issues/12764
